feat: webhooks and events pages (#77)
mia-platform/documentation: docs/development_suite/webhooks-and-events/events.mdx

---
id: events
title: Events
sidebar_label: Events
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

Events allow users to be notified when certain operations are performed in the Console.

## Event types

All the events triggered by the Console share some common properties:

| Property | Value | Description |
|-|-|-|
| `eventName` | `string` | The name of the event, usually in the form of `subject_action` |
| `eventTimestamp` | `number` | The timestamp of the event emission |
| `payload` | `object` | Contains the event-specific payload |

### `project_created`

This event will be triggered when a project is created successfully.

#### Event `payload`

| Property | Value | Description |
|-|-|-|

> | `tenantId` | `string` | The identifier of the tenant to which the project belongs |

Reviewer: ```suggestion
| `tenantId` | `string` | The identifier of the Company the Project belongs to |
```

> | `projectId` | `string` | The identifier of the tenant to which the project belongs |

Reviewer: ```suggestion
| `projectId` | `string` | The identifier of the newly created Project |
```

<Tabs>
<TabItem value="example" label="Example" default>

  ```json
  {
    "eventName": "project_created",
    "eventTimestamp": 123456,
    "payload": {
      "tenantId": "my-company-id",
      "projectId": "0000000000000001"
    }
  }
  ```

</TabItem>
<TabItem value="json-schema" label="JSON Schema">

  ```json
  {
    "additionalProperties": false,
    "properties": {
      "eventName": {
        "const": "project_created",
        "type": "string"
      },
      "eventTimestamp": {
        "type": "number"
      },
      "payload": {
        "additionalProperties": false,
        "properties": {
          "projectId": {
            "type": "string",
            "description": "The identifier of the project that was created"
          },
          "tenantId": {
            "type": "string",

>             "description": "The identifier of the tenant to which the project belongs"

Reviewer: ```suggestion
            "description": "The identifier of the Company the Project belongs to"
```

          }
        },
        "required": ["tenantId", "projectId"],
        "type": "object"
      }
    },
    "required": ["eventName", "eventTimestamp", "payload"],
    "type": "object"
  }
  ```

</TabItem>
</Tabs>

### `service_created`

This event will be triggered when a microservice is generated successfully from a marketplace template or example.

> **Note**: the event is triggered as soon as the service's repository is successfully created, hence there is no guarantee that the service will be found into the project configuration.

Reviewer: ```suggestion
:::note
The event is triggered as soon as the service's repository is successfully created, hence there is no guarantee that the service will be found into the project configuration, yet.
:::
```

---
id: events
title: Events
sidebar_label: Events
---

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

Events allow users to be notified when certain operations are performed in the Console.

## Event types

All the events triggered by the Console share some common properties:

| Property | Value | Description |
|-|-|-|
| `eventName` | `string` | The name of the event, usually in the form of `subject_action` |
| `eventTimestamp` | `number` | The timestamp of the event emission |
| `payload` | `object` | Contains the event-specific payload |

> ### `project_created`

Reviewer 2: Nei titoli userei la versione più leggibile, tipo " Project Creation" per questa e "Service Creation".
Tecnicamente so che i titoli sono giusti ma l'utente non ne avrà mai visibilità, anche su CMS ci sono le label più eplicative

| `projectId` | `string` | The identifier of the newly created Project |

<Tabs>
<TabItem value="example" label="Example" default>

  ```json
  {
    "eventName": "project_created",
    "eventTimestamp": 123456,
    "payload": {
      "tenantId": "my-company-id",
      "projectId": "0000000000000001"
    }
  }
  ```

</TabItem>
<TabItem value="json-schema" label="JSON Schema">

  ```json
  {
    "additionalProperties": false,
    "properties": {
      "eventName": {
        "const": "project_created",
        "type": "string"
      },
      "eventTimestamp": {
        "type": "number"
      },
      "payload": {
        "additionalProperties": false,
        "properties": {
          "projectId": {
            "type": "string",
            "description": "The identifier of the Project that was created"
          },
          "tenantId": {
            "type": "string",
            "description": "The identifier of the Company the Project belongs to"
          }
        },
        "required": ["tenantId", "projectId"],
        "type": "object"
      }
    },
    "required": ["eventName", "eventTimestamp", "payload"],
    "type": "object"
  }
  ```

</TabItem>
</Tabs>

> ### `service_created`

Reviewer 2: anche qui come sopra
